### DISCUSSION TITLE
Running parallel meep on ubuntu: "Meep must be configured/compiled with MPB for add_eigenmode_source"

### DISCUSSION THREAD

**Discussion by berzoll** [NanoComp/meep#2521] [General]
Hello,

I am trying to install parallel meep on an Ubuntu AWS machine. 
Following the installation instructions, I run the following commands:

"sudo apt-get install python3-meep-openmpi
sudo apt install python3-mpi4py"

I then install all the different packages needed with pip and run my script with "mpirun -np 1 python3 test.py" or "mpirun -np 1 python3 -u -m mpi4py test.py" (got only one core on this machine). The script runs fine until the line where I add the eigenmode source, which returns this output:

> ubuntu@ip-172-31-13-101:~$ mpirun -np 1 python3 -u -m mpi4py test.py
> Using MPI version 3.1, 1 processes
> Traceback (most recent call last):
>   File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
>     exec(code, run_globals)
>   File "/usr/lib/python3/dist-packages/mpi4py/__main__.py", line 7, in <module>
>     main()
>   File "/usr/lib/python3/dist-packages/mpi4py/run.py", line 198, in main
>     run_command_line(args)
>   File "/usr/lib/python3/dist-packages/mpi4py/run.py", line 47, in run_command_line
>     run_path(sys.argv[0], run_name='__main__')
>   File "/usr/lib/python3.10/runpy.py", line 289, in run_path
>     return _run_module_code(code, init_globals, run_name,
>   File "/usr/lib/python3.10/runpy.py", line 96, in _run_module_code
>     _run_code(code, mod_globals, init_globals,
>   File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
>     exec(code, run_globals)
>   File "testLinux.py", line 327, in <module>
>     opt.plot2D(True, ax = fig.add_subplot(111))
>   File "/usr/lib/python3/dist-packages/meep/adjoint/optimization_problem.py", line 397, in plot2D
>     self.prepare_forward_run()
>   File "/usr/lib/python3/dist-packages/meep/adjoint/optimization_problem.py", line 186, in prepare_forward_run
>     self.forward_monitors.append(m.register_monitors(self.frequencies))
>   File "/usr/lib/python3/dist-packages/meep/adjoint/objective.py", line 44, in register_monitors
>     self.normal_direction = self.monitor.normal_direction
>   File "/usr/lib/python3/dist-packages/meep/simulation.py", line 639, in normal_direction
>     return self.swigobj_attr('normal_direction')
>   File "/usr/lib/python3/dist-packages/meep/simulation.py", line 578, in swigobj_attr
>     self.swigobj = self.func(*self.args)
>   File "/usr/lib/python3/dist-packages/meep/simulation.py", line 2808, in _add_mode_monitor
>     self.init_sim()
>   File "/usr/lib/python3/dist-packages/meep/simulation.py", line 1935, in init_sim
>     self.add_source(s)
>   File "/usr/lib/python3/dist-packages/meep/simulation.py", line 2333, in add_source
>     add_eig_src(src.amp_func)
>   File "/usr/lib/python3/dist-packages/meep/__init__.py", line 3018, in add_eigenmode_source
>     return _meep.fields_add_eigenmode_source(self, *args)
> RuntimeError: meep: Meep must be configured/compiled with MPB for add_eigenmode_source
> 
> Elapsed run time = 0.8868 s
> --------------------------------------------------------------------------
> MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
> with errorcode 1.
> 
> NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
> You may or may not see output from other processes, depending on
> exactly when Open MPI kills them.
> --------------------------------------------------------------------------

What I am not doing correctly during the installation? I tried running "sudo apt-get install mpb-mpi" before installing meep, but to no use. Any hint how I could fix this?

Btw, seems that the AMI from Simpetus (http://www.simpetus.com/launchsims.html) does not exist anymore?

**Comment by smartalecH** [comment]
[As mentioned in the docs](https://meep.readthedocs.io/en/latest/Installation/#conda-packages), the preferred package manager solution is conda, which is the only one maintained by the meep developers.

**Comment by stevengj** [comment]
Note that Meep (even the MPI version) uses the serial version of MPB.  Note also that you probably need the `mpb-dev` package in order to get the header files etcetera.

**Comment by berzoll** [comment]
Thank you both! I managed to set everything  up with the conda packages just fine.
